assets/quick-order-list.js

```diff
@@ -160,23 +161,53 @@ class QuickOrderList extends HTMLElement {
     ];
   }
 
-  renderSections(parsedState) {
+  renderSections(parsedState, id) {
     this.getSectionsToRender().forEach((section => {
       const sectionElement = document.getElementById(section.id);
       if (sectionElement && sectionElement.parentElement && sectionElement.parentElement.classList.contains('drawer')) {
         parsedState.items.length > 0 ? sectionElement.parentElement.classList.remove('is-empty') : sectionElement.parentElement.classList.add('is-empty');
-
         setTimeout(() => {
           document.querySelector('#CartDrawer-Overlay').addEventListener('click', this.cart.close.bind(this.cart));
         });
       }
       const elementToReplace = sectionElement && sectionElement.querySelector(section.selector) ? sectionElement.querySelector(section.selector) : sectionElement;
       if (elementToReplace) {
-        elementToReplace.innerHTML =
-          this.getSectionInnerHTML(parsedState.sections[section.section], section.selector);
+        if (section.selector === '.js-contents' && id !== undefined) {
+          elementToReplace.querySelector(`#Variant-${id}`).innerHTML =
+          this.getSectionInnerHTML(parsedState.sections[section.section], `#Variant-${id}`);
+        } else {
+          elementToReplace.innerHTML = this.getSectionInnerHTML(parsedState.sections[section.section], section.selector);
+        }
       }
     }));
+    this.allInputsArray = Array.from(this.querySelectorAll('input[type="number"]'));
+    this.quickOrderListTable = this.querySelector('.quick-order-list__table');
+    this.quickOrderListTable.addEventListener('focusin', this.switchVarints.bind(this));
```

Comment:
That's a fair question. The first time we run those lines of code for the initial scenario when you just load a page. It's happening inside of the `constructor( )` of the QuickOrderList element. Since we don't re-render the whole QuickOrderList element anymore the code inside `constructor( )` won't be run again. At the same time we do changes in a row which means that for DOM the initial `this.allInputsArray` doesn't exist anymore. Previously, it would be re-built in `constructor( )` on reload step but now we just need to rebuild it somewhere else. I hope it makes sense.

Reply:
@eugenekasimov Makes sense! Can we add this to a function to avoid running the same lines twice? This way we can call the function